### ISSUE TITLE
Handle destructured import of a test modifier

### ISSUE THREAD

**Issue by sindresorhus** [avajs/eslint-plugin-ava#121]
[<img alt="Issuehunt badges" src="https://issuehunt.io/r/avajs/eslint-plugin-ava/issues/121/badge.svg" />](https://issuehunt.io/r/avajs/eslint-plugin-ava/issues/121)

From https://github.com/sindresorhus/conf/issues/2#issuecomment-227897259

I think we should handle the following cases:

``` js
import {serial as test} from 'ava';
```

``` js
import {serial} from 'ava';
```

It's a useful pattern to make a test file with only serial tests.

> There is a $80.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/avajs/eslint-plugin-ava/issues/121).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/avajs/eslint-plugin-ava/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

**Comment by jfmengels** [comment]
Are there any other methods that are aliases to `test.XXX`?
If I do
```js
import {cb as test} from 'ava';
test(...);
```
is that equivalent to
```js
import test from 'ava';
test.cb(...);
```
or is `serial` a special one?

**Comment by jamestalmage** [comment]
All methods can be destructured and will work as expected, all the following are valid:

```
{cb as test}
{serial as test}
{cb as test}
```

You could even try something like this for a mocha-like interface:

```js
import {beforeEach, afterEach, cb as it} from 'ava';

beforeEach(t => {...});

it('does things', t => {...});
```

Most of our rules should try and support all of those. We can update the `use-test` rule to warn against any particular patterns we don't like.

**Comment by jfmengels** [comment]
An idea for an other time, but maybe we could have an option in `use-test` that forbids destructuring of `test`, if users want to stick to the explicit `test`.

**Comment by jfmengels** [comment]
By the way, this looks a lot like what I did in [eslint-plugin-lodash-fp](https://github.com/jfmengels/eslint-plugin-lodash-fp/blob/master/rules/core/enhance.js#L30-L64). Starting to think we could maybe make a separate "enhancer" for this using [enhance-visitor](https://github.com/jfmengels/enhance-visitors).

**Comment by jamestalmage** [comment]
Much of the static-analysis work is already complete in [`babel-plugin-ava-test-data`](https://github.com/avajs/babel-plugin-ava-test-data), so it should just be a matter of porting for the eslint API.

**Comment by IssuehuntBot** [comment]
@issuehunt has funded $80.00 to this issue.

---
- Submit pull request via [IssueHunt](https://issuehunt.io/repos/49722492/issues/121) to receive this reward.
- Want to contribute? Chip in to this issue via [IssueHunt](https://issuehunt.io/repos/49722492/issues/121).
- Checkout the [IssueHunt Issue Explorer](https://issuehunt.io/issues) to see more funded issues.
- Need help from developers? [Add your repository](https://issuehunt.io/r/new) on IssueHunt to raise funds.
